Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionPathsTask.cs

```diff
@@ -0,0 +1,117 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Emby.Server.Implementations.Collections;
+using MediaBrowser.Controller.Collections;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Entities.Movies;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.IO;
+using MediaBrowser.Model.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Emby.Server.Implementations.ScheduledTasks.Tasks;
+
+/// <summary>
+/// Deletes Path references from collections that no longer exists.
+/// </summary>
+public class CleanupCollectionPathsTask : IScheduledTask
+{
+    private readonly ILocalizationManager _localization;
+    private readonly ICollectionManager _collectionManager;
+    private readonly ILogger<CleanupCollectionPathsTask> _logger;
+    private readonly IProviderManager _providerManager;
+    private readonly IFileSystem _fileSystem;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CleanupCollectionPathsTask"/> class.
+    /// </summary>
+    /// <param name="localization">Instance of the <see cref="ILocalizationManager"/> interface.</param>
+    /// <param name="collectionManager">Instance of the <see cref="ICollectionManager"/> interface.</param>
+    /// <param name="logger">The logger.</param>
+    /// <param name="providerManager">The provider manager.</param>
+    /// <param name="fileSystem">The filesystem.</param>
+    public CleanupCollectionPathsTask(
+        ILocalizationManager localization,
+        ICollectionManager collectionManager,
+        ILogger<CleanupCollectionPathsTask> logger,
+        IProviderManager providerManager,
+        IFileSystem fileSystem)
+    {
+        _localization = localization;
+        _collectionManager = collectionManager;
+        _logger = logger;
+        _providerManager = providerManager;
+        _fileSystem = fileSystem;
+    }
+
+    /// <inheritdoc />
+    public string Name => _localization.GetLocalizedString("TaskCleanCollections");
+
+    /// <inheritdoc />
+    public string Key => "CleanCollections";
+
+    /// <inheritdoc />
+    public string Description => _localization.GetLocalizedString("TaskCleanCollectionsDescription");
+
+    /// <inheritdoc />
+    public string Category => _localization.GetLocalizedString("TasksMaintenanceCategory");
+
+    /// <inheritdoc />
+    public async Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellationToken)
+    {
+        var collectionsFolder = await _collectionManager.GetCollectionsFolder(false).ConfigureAwait(false);
+        if (collectionsFolder is null)
+        {
+            _logger.LogInformation("There is no collection folder to be found.");
+            return;
+        }
+
+        var collections = collectionsFolder.Children.OfType<BoxSet>().ToArray();
+        _logger.LogTrace("Found {CollectionLength} Boxsets.", collections.Length);
```

Comment:
```suggestion
        _logger.LogTrace("Found {CollectionLength} Boxsets", collections.Length);
```

`Trace` includes very verbose logging, I would stick to `Debug`